addons/web_editor/static/src/js/wysiwyg/widgets/link.js

```diff
@@ -482,6 +482,7 @@ const Link = Widget.extend({
         let isLink = value.indexOf('@') < 0;
         this.$('input[name="is_new_window"]').closest('.form-group').toggleClass('d-none', !isLink);
         this.$('.o_strip_domain').toggleClass('d-none', value.indexOf(window.location.origin) !== 0);
+        this._adaptPreview();
```

Comment:
You can't do that at each input without preventing the observer to function, otherwise, each keystroke will create a new step.

Undoing a step where only the `href` attribute changes is invisible for the user and will make the undo seem to not working properly.